surfactant/filetypeid/id_magic.py

```diff
@@ -208,6 +208,36 @@ def identify_file_type(filepath: str) -> Optional[str]:
                 if cm == 8:
                     if (cmf * 256 + flg) % 31 == 0:
                         return "ZLIB"
+            # cpio:
+            # https://www.systutorials.com/docs/linux/man/5-cpio/
+            if int.from_bytes(magic_bytes[:2], byteorder="big", signed=False) == 0o70707:
+                return "CPIO_BIN big"
+            if int.from_bytes(magic_bytes[:2], byteorder="little", signed=False) == 0o70707:
+                return "CPIO_BIN little"
+            if magic_bytes[:6] == b"070707":
+                return "CPIO_CHAR_OLD"
+            if magic_bytes[:6] == b"070701":
+                return "CPIO_ASCII_NEW"
+            # zstd:
+            # https://datatracker.ietf.org/doc/html/rfc8878
+            if magic_bytes[:4] == b"\x28\xb5\x2f\xfd":
+                return "ZSTANDARD"
+            if magic_bytes[:4] == b"\x37\xa4\x30\xec":
+                return "ZSTANDARD_DICTIONARY"
+            # iso:
+            # https://www.garykessler.net/library/file_sigs.html
+            f.seek(0)
+            iso_bytes = f.read(0x9001 + 5)
+            for offset in (0x8001, 0x8801, 0x9001):
+                print(iso_bytes[offset : offset + 5])
+                if iso_bytes[offset : offset + 5] == b"CD001":
+                    return "ISO_9660_CD"
```

Comment:
```suggestion
            for offset in (0x8001, 0x8801, 0x9001):
                f.seek(offset) // seeks to absolute position by default
                iso_bytes = f.read(5)
                if iso_bytes == b"CD001":
                    return "ISO_9660_CD"
            f.seek(0) // reset fp offset to start of file for subsequent reads
```
I think we can do some more seeking to minimize the size